### ISSUE TITLE
Catch unauthenticated errors

### ISSUE THREAD

**Issue by MarkEdmondson1234** [andodet/googlePubsubR#10]
I'm using `googlePubsubR` in the latest `googleCloudRunner` release to set-up some PubSub topics, it worked locally but then on the package checks for Cloud Build got a strange error `Error: API returned: You have passed an invalid argument to the service (argument=).` which now I think of it must be as the environment arguments aren't set in the Cloud Build.  

I will fix that, but the error message could have been more helpful, perhaps catching if those variables are empty.

**Comment by andodet** [comment]
That looks like an API call was made having an empty string as one of the arguments. Do you have a repr or a `googleCloudRunner` feature branch I can look at?

**Comment by MarkEdmondson1234** [comment]
It was this code:

https://github.com/MarkEdmondson1234/googleCloudRunner/blob/ca67ceef4f82ad77ebf72665d7a510c0258cfafb/R/deploy.R#L142-L152

```r
        myMessage("Creating PubSub topic:", topic_basename, level = 3)
        # Create a pubsub topic
        topic_created <- tryCatch(
          googlePubsubR::topics_create(topic_basename),
            error = function(err){
              stop("Could not create topic:",
                   topic_basename,
                   err$message,
                   call. = FALSE)
            })
      }
```

...and was fixed once I specified the environment argument.

**Comment by andodet** [comment]
The culprit is inceed `GCP_PROJECT_ID` that resolves to an empty string when the env variable is not set in `0.0.2`. Switching the getter/setter pattern you suggested in #11 will guard against this. Thanks a lot for your input

**Comment by andodet** [comment]
I've just merged #13 into `dev`. Cases in which GCP projectId is set to an empty string (e.g unset env `GCP_PROJECT`) will now result in an error. This should prevent the behaviour you've observed in which the API call was fired regardless.
It would be perfect if you could give the `dev` branch a spin to confirm this has resolved before merging to `master` and resubmitting to CRAN. Thanks again for all the feedback :raised_hands:.

**Comment by MarkEdmondson1234** [comment]
I've installed it and will let you know if I see it again :)

**Comment by andodet** [comment]
No idea if that eventually solved your issue, happy to bundle that change alongside other fixes for a `0.0.3` release in early Jan.

**Comment by MarkEdmondson1234** [comment]
It looks good, please do include :)
